android/sound.cpp

```diff
@@ -306,13 +307,14 @@ oboe::DataCallbackResult CSound::onAudioOutput ( oboe::AudioStream* oboeStream,
     // According to the format that we've set on initialization, audioData
     // is an array of int16_t
     //
+    int16_t* intData = static_cast<int16_t*> ( audioData );
 
-    memcpy ( audioData, outBuffer.data(), count * sizeof ( int16_t ) );
+    memcpy ( intData, outBuffer.data(), sizeof ( int16_t ) * count );
```

Comment:
Note that by putting `sizeof(int16_t)` first in the expression, it should correctly avoid overflow without `count` having to be of type `std::size_t`. Similarly a few lines further down.